WooCommerce/UITestsFoundation/Screens/Orders/UnifiedOrderScreen.swift

```diff
@@ -236,4 +236,26 @@ public final class UnifiedOrderScreen: ScreenObject {
 
         return self
     }
+
+    /// Checks the screen for existence of all products, checking each name.
+    /// - Parameter productNames: Array of product names to check.
+    /// - Returns: Unified Order screen object.
+    @discardableResult
+    public func checkForExistingProducts(byName productNames: [String]) throws -> UnifiedOrderScreen {
+        for productName in productNames {
+            let productNamePredicate = NSPredicate(format: "label MATCHES %@", productName)
+            XCTAssertTrue(app.staticTexts.containing(productNamePredicate).element.exists)
+        }
+
+        return self
+    }
+
+    /// Finishes Order Editing process
+    /// - Returns: Single Order Screen object.
+    @discardableResult
```

Comment:
```suggestion
```

same as above, think this can be removed.

Reply:
@jostnes as I was removing this and the previous `@discardableResult`, I realized that we have a lot of `@discardableResult`s throughout `UnifiedOrderScreen` and `SingleOrderScreen`.

My understanding is that  `@discardableResult` is used to suppress the warning when we have a `return` that we won't actually use. 
In our UI tests, much of the time we `return` a screen, and then the screen in question is displayed. So in those cases, our result is not discardable - we don't expect to get a warning that we aren't using the return, so we should not suppress such a warning.
In a few cases, we `return self`, in which case we do not change screens. In this case, we do not use the `return`, and we do expect to see a warning about this. Since we know why the warning appears, and we are certain this is the behavior we want, we can suppress the warning by using `@discardableResult`.

Am I understanding that correctly?

If so, I can go ahead and remove the unnecessary `@discardableResult`s from `UnifiedOrderScreen` and `SingleOrderScreen` - those for functions that do return a different screen.

Reply:
Yup, that's correct - `@discardableResult` is used to suppress the warning when we have a return that we won't actually use. 

We return the screen value because as we are chaining the tests, each step in the chain is expecting a certain screen. Those `return` values are used to ensure that the test knows at that point, it's on which screen. This is for both cases when we return a different screen or `self` (which is equivalent to returning the screen it was originally on). In the cases where the method is the last step for the test, the chain stops, and the returned screen is not used anymore. In this case, we can use `@discardableResult` to suppress the warning (or we can even remove the return value since we know it's not going to be used and there are usually assertions in those methods)

So yup, you can go ahead and remove all the unnecessary `@discardableResult` from the screen. Thanks for cleaning it up!

Reply:
Adding my point of view: while removing `@discardableResult` won't trigger a warning in the current code, I think it serves additional documentation purpose. It means that annotated function is primarily used for its "side effect" and the return value is just a handy reference if you need it.

Reply:
Thanks for the discussion! I've removed `@discardableResult` in cases where we return a different screen, and left it for methods that return `self`.